### DISCUSSION TITLE
Global Variables

### DISCUSSION THREAD

**Discussion by itizarsa** [scalar/scalar#1570] [General]
We have a header required to be sent on most of the requests, currently we have to enter the value for the header on each operation. Is there a way to define a global variable similar to AuthZ & and provide it in the example field of the header parameter?

**Comment by marclave** [comment]
hi @itizarsa sorry on the late reply!

would you want to add these variables to the openapi spec?

**Reply by itizarsa** [reply]
@marclave I'm not aware how to add this in openapi spec, if that's possible I will do that

**Comment by TimoGlastra** [comment]
We'd also be interested in this. Almost all our endpoints require a `{projectId}` to be passed, but not all. For this reason we can't use the current server variables and use path replacement, as the base doesn't include the projectId. But this means that for every request the projectId would have to be filled in. 

It would be really useful if we can define some global variables, or if the server variables could be used for request headers/params as well. Then the user could select the projectId at the top, and it would automatically be filled in for all requests that require it.

**Reply by TimoGlastra** [reply]
You can see our API reference here: https://paradym.id/reference
